examples/exampleSetup.hpp

```diff
@@ -118,49 +118,72 @@ namespace example {
         return flux;
     }
 
-    Opm::FlowDiagnostics::Toolbox
-    initialiseFlowDiagnostics(const Opm::ECLGraph& G,
-                              const std::vector<Opm::ECLWellSolution::WellData>& well_fluxes,
-                              const int step)
+    template <class WellFluxes>
+    Opm::FlowDiagnostics::CellSetValues
+    extractWellFlows(const Opm::ECLGraph& G,
+                     const WellFluxes&    well_fluxes)
     {
-        const auto connGraph = Opm::FlowDiagnostics::
-            ConnectivityGraph{ static_cast<int>(G.numCells()),
-                               G.neighbours() };
-
-        using FDT = Opm::FlowDiagnostics::Toolbox;
-
-        auto fl = extractFluxField(G, step);
-        const size_t num_conn = fl.numConnections();
-        const size_t num_phases = fl.numPhases();
-        for (size_t conn = 0; conn < num_conn; ++conn) {
-            using Co = Opm::FlowDiagnostics::ConnectionValues::ConnID;
-            using Ph = Opm::FlowDiagnostics::ConnectionValues::PhaseID;
-            for (size_t phase = 0; phase < num_phases; ++phase) {
-                fl(Co{conn}, Ph{phase}) /= 86400; // HACK! converting to SI.
-            }
-        }
-
         Opm::FlowDiagnostics::CellSetValues inflow;
         for (const auto& well : well_fluxes) {
             for (const auto& completion : well.completions) {
                 const int grid_index = completion.grid_index;
                 const auto& ijk = completion.ijk;
                 const int cell_index = G.activeCell(ijk, grid_index);
-                inflow.addCellValue(cell_index, completion.reservoir_inflow_rate);
+                if (cell_index >= 0) {
+                    inflow.addCellValue(cell_index, completion.reservoir_inflow_rate);
+                }
+            }
+        }
+
+        return inflow;
+    }
+
+    namespace Hack {
```

Comment:
The reason this is a hack is that:
 - it does not check actual units used (but assumes cubic meters/day)
 - it does not use the Opm unit system constants

This is fixable: instead of the current solution, we should implement proper unit usage in ECLGraph (where the context is available to choose the correct unit for any unit system in use).